chain-abci/src/storage/mod.rs

```diff
@@ -14,7 +15,92 @@ use chain_tx_validation::{
     verify_node_join, verify_unbonding, verify_unjailed, verify_unjailing,
     witness::verify_tx_recover_address, ChainInfo, Error, NodeChecker,
 };
-use enclave_protocol::{IntraEnclaveRequest, IntraEnclaveResponseOk, SealedLog, VerifyOk};
+use enclave_protocol::{IntraEnclaveRequest, IntraEnclaveResponseOk, SealedLog};
+
+/// fee: Written into block result events
+/// spend_utxo: Modify UTxO storage
+/// create_utxo: Write into UTxO storage
+/// sealed_log: Write into storage
+/// deposit/withdraw: Modify staking state and related validator state structure
+#[derive(Debug, Clone)]
+pub enum TxEnclaveAction {
+    Transfer {
+        fee: Fee,
+        spend_utxo: Vec<TxoPointer>,
+        create_utxo: TxoIndex,
```

Comment:
`TxoIndex` type alias is a bit confusing name here, because here, it's a TX output length

Reply:
(i.e. number of outputs)

Reply:
Yeah, it's kind of like `usize`, which used both as to length and index.
Maybe change to `TxoSize` is more similar to the naming convention of `usize`?

Reply:
yep https://github.com/crypto-com/chain/issues/1223